plugins/source/aws/resources/services/cloudwatchlogs/metric_filters.go

```diff
@@ -13,18 +13,10 @@ func MetricFilters() *schema.Table {
 		Description: `https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_MetricFilter.html`,
 		Resolver:    fetchCloudwatchlogsMetricFilters,
 		Multiplex:   client.ServiceAccountRegionMultiplexer("logs"),
-		Transform:   transformers.TransformWithStruct(&types.MetricFilter{}),
+		Transform:   transformers.TransformWithStruct(&types.MetricFilter{}, transformers.WithPrimaryKeys("LogGroupName", "FilterName")),
 		Columns: []schema.Column{
-			client.DefaultAccountIDColumn(false),
-			client.DefaultRegionColumn(false),
-			{
-				Name:     "arn",
-				Type:     schema.TypeString,
-				Resolver: resolveMetricFilterArn,
-				CreationOptions: schema.ColumnCreationOptions{
-					PrimaryKey: true,
-				},
-			},
+			client.DefaultAccountIDColumn(true),
+			client.DefaultRegionColumn(true),
```

Comment:
Maybe convert the `LogGroupName` to an arn then we can use just those 2 columns (LogGroupARN and FilterName) for the PK... That way it also would make it easier to query and do joins between `aws_cloudwatchlogs_metric_filters` and `aws_cloudwatchlogs_log_groups`

Reply:
but yeah overall seems better, will do

Reply:
You would have to join on `account_id`, `region` and `log_group_name` because the names can repeat across regions... I have actually used that pattern in the past (putting the same log group name in every production region)

Reply:
It my view it is a potential "foot gun"... People won't always think about regional naming constraints and then they will blame CQ for duplicating data

Reply:
Fixed in [`52376a1` (#8061)](https://github.com/cloudquery/cloudquery/pull/8061/commits/52376a1165d66143663baf209d2d740132cec8cc)